src/main.rs

```diff
@@ -31,15 +32,33 @@ use {
 };
 
 #[allow(unused_mut)]
-#[tokio::main]
-async fn main() -> Result<()> {
+fn main() -> Result<()> {
     let cli = DftArgs::parse();
 
+    let state = state::initialize(cli.config_path());
+
+    let cpus = num_cpus::get();
+    let main_threads = if state.config.execution.dedicated_executor_enabled {
+        // Just for IO
+        (cpus as f64 * (1.0 - state.config.execution.dedicated_executor_threads_percent)) as usize
+    } else {
+        cpus
+    };
```

Comment:
Maybe i could just build the threads logic into the config (i.e. have fields for main / dedicated executor threads) that are computed at config load time.

Reply:
FWIW we found it is ok to slightly over commit the CPUs (like if you have 8 CPUs total, it is often fine to have an 8 CPUs in the dedicated worker, and 1 CPU). 

For most workloads, leaving a single CPU to process network requests is totally fine (e.g. with 8 CPUs total, using 7 in the dedicated executor is fine as it will leave 1 entire CPU to do most of the real work)

Reply:
That makes a lot of sense